python/foglamp/services/core/api/service.py

```diff
@@ -95,72 +96,105 @@ def get_service_records():
         is_enabled = True if ((type(enabled) is str and enabled.lower() in ['t', 'true']) or (
             (type(enabled) is bool and enabled is True))) else False
 
-        storage = connect.get_storage_async()
+        # ---------------------------------------- Sanity checks
+        # Check if a valid plugin has been provided
+        try:
+            # "plugin_module_path" is fixed by design. It is MANDATORY to keep the plugin in the exactly similar named
+            # folder, within the plugin_module_path.
+            plugin_module_path = "foglamp.plugins.south" if service_type == 'south' else \
+                                 "foglamp.plugins.north" if service_type == 'north' else None
```

Comment:
it should NEVER be none! better to check `service_type in ['south', 'north']` and then `plugin_module_path = "foglamp.plugins.{}".format(service_type)`